applications/FluidDynamicsApplication/custom_conditions/monolithic_wall_condition.h

```diff
@@ -512,7 +512,7 @@ class KRATOS_API(FLUID_DYNAMICS_APPLICATION) MonolithicWallCondition : public Co
         for(size_t itNode = 0; itNode < rGeometry.PointsNumber(); ++itNode)
         {
             const NodeType& rConstNode = rGeometry[itNode];
-            const double y = rConstNode.GetValue(Y_WALL); // wall distance to use in stress calculation
+            const double y = rConstNode.FastGetSolutionStepValue(DISTANCE); // wall distance to use in stress calculation
```

Comment:
This is a big behavior change and forces all users of the `MonolithicWallCondition` to allocate distance on nodes...
I would very much prefer if we kept the current behavior and assigned `GetValue(Y_WALL)` to be equal to `FastGetSolutionStepValue(DISTANCE)` in an initialization loop somewhere.

Reply:
They do not, I was using the historical variables because variational distance calculation process need them to be historical variables. This modification will be moved to RANSModellingApplication as @jcotela  suggested :)

Reply:
I haven't taken a look at it. I think it would also require to port the calculation_distance_processes to non-historical as well.

Is it possible to use the builder and solver (used in the variational distance process) with non-historical variables though?

Reply:
hm good question, I am actually quite sure that it does not
what is your dof in that case? `DISTANCE`?

Reply:
Yes. The DOF is the `DISTANCE` variable.